glean-core/ios/Glean/Scheduler/MetricsPingScheduler.swift

```diff
@@ -166,13 +164,11 @@ class MetricsPingScheduler {
             // `Dispatchers` serial execution queue, before any other enqueued task. For more
             // context, see bug 1604861 and the implementation of
             // `collectPingAndReschedule`.
-            Dispatchers.shared.serialOperationQueue.addOperation {
-                self.collectPingAndReschedule(
-                    now,
-                    startupPing: true,
-                    reason: .overdue
-                )
-            }
+            self.collectPingAndReschedule(
```

Comment:
This negates the whole block of comments above it.
Can you clarify why that is fine? Possibly adding that as a comment.

I suspect it is because now it's happening in the async startup, but there's no guarantuee other than us calling it there, or is there?

Reply:
This is because this part of `schedule` no longer needs to be called off the main thread because `initialize` is called off the main thread, and since this is being called synchronously (in the background thread) before we flush the queued tasks, it's guaranteed to be called before any metric API adds data when the queue is flushed. 

I'll update the comment to reflect this.